sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -142,17 +142,18 @@ class AstBuilder extends DataTypeAstBuilder
   }
 
   override def visitSingleCompoundStatement(ctx: SingleCompoundStatementContext): CompoundBody = {
-    visit(ctx.beginEndCompoundBlock()).asInstanceOf[CompoundBody]
+    val seenLabels = Set[String]()
```

Comment:
Yes this is the implementation I was expecting. Can we do a bit more encapsulation? e.g.
```
val labelCtx = new ScritpingLabelContext()
visitBeginEndCompoundBlockImpl...
...
class ScritpingLabelContext {
  private val seenLabels = Set[String]()
  
  def newLabel(label: String)...
  ...
}
``

Reply:
@cloud-fan done.